ppa/archive/solr.py

```diff
@@ -123,11 +123,10 @@ def query_opts(self):
             # use set to ensure we don't duplicate a filter
             qs_copy.filter_qs = list(set(qs_copy.filter_qs))
 
-            # if not searching within a group, collapse reprints/editions
-            # @NOTE: This following code may not be necessary?
-            # if not qs_copy.within_cluster_id:
-            # (not expanding, since we only display the first)
-            # qs_copy = qs_copy.filter("{!collapse field=cluster_id_s}")
+            # If not searching within a group, still collapse reprints/editions
+            # Remember that cluster_id_s is now defined as `str(self.cluster) if self.cluster else index_id` in models.py.
+            # So collapsing by "cluster" id implicitly includes works with no cluster id set.
+            qs_copy = qs_copy.filter("{!collapse field=cluster_id_s}")
```

Comment:
Was thinking that this was redundant code, but I see that we bail out early if there are no keyword search terms, since there's a lot of logic we can skip. 

Could we move the `collapse_on` and `collapse_filter` logic currently on lines 160-166 and rely on that for both cases where we collapse? 

Actually, I think it should work to apply the collapse once outside of the if statement, since we always want it and the logic should be the same. Please try that — but if it complicates things too much or causes problems, you can back out to something more like this (but reusing the common variables).